interpreter/cling/lib/Interpreter/ValuePrinter.cpp

```diff
@@ -902,33 +902,33 @@ static std::string printUnpackedClingValue(const Value &V) {
       = llvm::dyn_cast<clang::BuiltinType>(Td.getCanonicalType().getTypePtr())) {
     switch (BT->getKind()) {
       case clang::BuiltinType::Bool:
-        return executePrintValue<bool>(V, V.getLL());
+        return executePrintValue<bool>(V, V.getLongLong());
 
       case clang::BuiltinType::Char_S:
-        return executePrintValue<signed char>(V, V.getLL());
```

Comment:
it feels like both the old and new code would fail on big endian.  @Axel-Naumann Shouldn't those be `V.getAs<signed char>` and the likes?

Reply:
I agree. I looked how to get rid of the entire dispatch function and I failed. I still think we can remove it altogether but need to stare more at this code...

Reply:
Should we open an issue to check on this later?

Reply:
It is a functionality issue.  Does Value works or does not work on big endian?  I think that fixing/checking this will be a pre-requisite to uploading it to llvm.

Reply:
I do not have a reasonable big endian machine unfortunately, neither CERN. I tried some WINE emulation that was super slow which had issues elsewhere... Not sure how to make progress here. I am pretty sure if something does not work when we land it in llvm their bots will catch it and then we can contact the bot owner to give us access.

Reply:
humm ... if we (essentially/indirectly) do (to set the value):
```
*((short*)someaddress) = value
```
and (here) do:
```
(short) *(long long*)sameaddress)
```
the result has to be wrong on either big endian or little endian unless (with is likely) I am wrong on the way union are implemented.

To convince ourselves whether the code is 'obviously' wrong or just 'maybe' , maybe we can use Compiler explorer.

Reply:
Yes, this was discussed in length in the previous PR or somewhere in this PR. I don’t remember anymore but we beat this topic to death.

Reply:
Thanks for fixing this.